aiida/backends/managers/access.py

```diff
@@ -0,0 +1,259 @@
+            self._read_from_file(filepath)
+            return
+
+        if pid is None:
+            pid = os.getpid()
+
+        self._pid = pid
+        process_obj = psutil.Process(pid)
+        process_cmd = process_obj.cmdline()
+        process_ctm = time.localtime(process_obj.create_time())
+        process_ctm = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(process_obj.create_time()))
+
+        self._data = {'cmd': process_cmd, 'ctime': process_ctm}
+
+    def write_to_file(self, filepath):
+        """Writes the info of the process to a file (filepath)"""
+        json_data = {'pid': self._pid, 'data': self._data}
+        with open(filepath, 'w') as json_file:  # pylint: disable=unspecified-encoding
+            json.dump(json_data, json_file)
+
+    def _read_from_file(self, filepath):
+        """Reads the info of the process from a file (filepath)"""
+        with open(filepath, 'r') as json_file:  # pylint: disable=unspecified-encoding
+            json_data = json.load(json_file)
+        self._pid = json_data['pid']
+        self._data = json_data['data']
+
+    def __eq__(self, other):
+        """Define equivalency"""
+        if isinstance(other, self.__class__):
+            return (self.pid == other.pid) and (self.cmd == other.cmd)
+        return False
+
+    @property
+    def pid(self):
+        """pid"""
+        return self._pid
+
+    @property
+    def cmd(self):
+        """cmd"""
+        return self._data['cmd']
+
+
+class AccessManager():
+    """Class to manage and keep track of processes accessing and locking the profile."""
+
+    def __init__(self, profile=None):
+        """Initialization process for the access manager object.
+
+        :param profile:
+            profile to manage the access of (loaded profile by default).
+        """
+        from aiida.manage.configuration.settings import ACCESS_CONTROL_DIR
+        from aiida.manage.manager import get_manager
+
+        if profile is None:
+            profile = get_manager().get_profile()
+
+        self._temp_prefix = 'request_'
+        self._profile = profile
+        profile_access_path = ACCESS_CONTROL_DIR / profile.name
+        profile_access_path.mkdir(exist_ok=True)
+
+        self._path_to_tracked = profile_access_path / 'tracked'
+        self._path_to_tracked.mkdir(exist_ok=True)
+
+        self._path_to_locking = profile_access_path / 'locking'
+        self._path_to_locking.mkdir(exist_ok=True)
+
+    def get_processes_recorded_in(self, basepath, ignore_self=True):
+        """Returns all processes tracked in a given path
+
+        :param basepath:
+            base path in which the records are being stored.
+
+        :param ignore_self:
+            flag to avoid returning the process that is requesting the information in the output
+            (enabled by default).
+        """
+        process_datadicts = {}
+
+        for filename in os.listdir(basepath):
+            if not filename.startswith(self._temp_prefix):
```

Comment:
I would actually also check that the file ends with `.pid` (this should also be a class variable, reused everywhere and not hardcoded). This avoids that if there are other files these create problems (e.g. files ending in `~` if someone edited with and editor, files like `.gitignore` for some reason put there, ...)